### DISCUSSION TITLE
Setting up the plugin for a typescript web project

### DISCUSSION THREAD

**Discussion by baretd** [green-code-initiative/creedengo-javascript#27] [General]
Hello,

1. Install

[Here](https://github.com/green-code-initiative/ecoCode-javascript/pkgs/npm/ecocode-eslint-plugin) 

You can see several plugin installation commands.
First one 
![image](https://github.com/green-code-initiative/ecoCode-javascript/assets/63282503/27a6c2be-6728-48ce-99de-4ffcfbbdeaae)
Second one 
![image](https://github.com/green-code-initiative/ecoCode-javascript/assets/63282503/b8c24f3f-24ef-4dc0-987c-20fb0f5240cb)

What's the difference between the two packages?

2. Usage
In relation to the [collective's book](https://www.greenit.fr/), only these rules have been implemented  : 
![image](https://github.com/green-code-initiative/ecoCode-javascript/assets/63282503/ee46b24a-708c-4a73-a247-1f80f27acf9d)
Is this correct?

Thank you for your work

**Comment by utarwyn** [comment]
Hello @baretd,
I apologize for the lack of documentation on the ESLint project. There's still a way to go to explain everything. Thanks for your feedback!

### Packages explanation

There are in fact two different packages, but they contain **the same content** (both are created at the same time with same versions):

- [@ecocode/eslint-plugin](https://www.npmjs.com/package/@ecocode/eslint-plugin): the "default" package accessible from npmjs.org
- [@green-code-initiative/ecocode-eslint-plugin](https://github.com/green-code-initiative/ecoCode-javascript/pkgs/npm/ecocode-eslint-plugin): the package for those who use GitHub as a package registry ([more info here](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry))

You can use either one or the other, depending on your project configuration. 

### Implemented rules

More and more rules are added as new versions are released. You can find two tables listing the rules in two different states of the project:

- table on [npmjs.org page of the project](https://www.npmjs.com/package/@ecocode/eslint-plugin) lists the rules of the latest published and available version (5 at the moment of writing - v0.2.0)
- table on the [GitHub repository](https://github.com/green-code-initiative/ecoCode-javascript/tree/main/eslint-plugin#-rules) lists the rules that will be included in the next version of the plugin (10 right now)

I hope these answers will give you a clearer picture! 😉

**Reply by dedece35** [reply]
@utarwyn,
do you think this concise explanation could be on technical documentation ?

**Reply by utarwyn** [reply]
@dedece35
yes I have planned to update the README to add these informations

**Reply by baretd** [reply]
Thanks for the reply!
